### ISSUE TITLE
SL3-1-5: Developing Ev Encoder that is compatible with SONAR-SLT backbone

### ISSUE THREAD

**Issue by akraradets** [AIT-brainlab/SignLanguage#37]
# Issue

Develop an Event Visual Encoder for Pipeline B that replaces SONAR-SLT's RGB-based visual front-end (ViT + VideoMAE) with an event-based encoder inspired by EvSign. The encoder takes voxel grid representations of synthesized DVS events and produces a visual embedding z aligned with the SONAR text encoder space and SONAR Decoder generate the text output. 

Pipeline: Voxel Grid → Sparse Backbone → Local Token Fusion → Temporal Pooling → Projection → z
Depends on: voxel grid preprocessing (events.h5 → .npz, B=5)

## To Do

- [x] Data Pipeline & Dataloader — voxel .npz + PHOENIX-2014-T CSV, augmentation ( 8hr )
- [x] Training Script — Adam + cosine LR, batch size 2–8, monitor cosine sim + BLEU on dev split ( 8hr )
- [x] Evaluation — BLEU-1/2/3/4 + ROUGE-L on test split, compare vs. RGB baseline ( 6hr )

**Comment by phuepwint-thwe** [comment]
### Result Table

| Run | Split | BLEU-4 | ROUGE-L | Cosine z-s | MSE z-s | Takeaway |
|---|---:|---:|---:|---:|---:|---|
| v0 wrong-loss | Test | 3.53 | 14.09 | 0.4727 | 7.4e-05 | Baseline failed with mode collapse |
| v1 corrected-loss | Dev | 4.71 | 16.17 | 0.4468 | 7.3e-05 | Slightly better, but still mode collapse |

### Plots

#### Joint Loss

<img width="1600" height="1000" alt="Image" src="https://github.com/user-attachments/assets/37da926a-e6de-45f1-8d18-e8146fada1ad" />

#### MSE Alignment

<img width="1600" height="1000" alt="Image" src="https://github.com/user-attachments/assets/06aefe07-5162-40b7-96b9-21262bb3f751" />


#### Validation Cosine Alignment

<img width="1600" height="1000" alt="Image" src="https://github.com/user-attachments/assets/a6dd04bd-a603-4062-88a0-0882f6e320d9" />


#### Decoder CE

<img width="1600" height="1000" alt="Image" src="https://github.com/user-attachments/assets/ff233952-09d5-41d9-aa84-8879c5258c9d" />



#### Decoder Anchor CE

<img width="1600" height="1000" alt="Image" src="https://github.com/user-attachments/assets/aae35d7a-26ed-478b-b735-52c953d26c02" />


| Plot | Takeaway |
|---|---|
| `loss.png` | Validation loss is best around epoch 17, then overfitting starts. |
| `mse_alignment.png` | MSE decreases, so `z` moves closer to SONAR text embedding space. |
| `val_cosine_alignment.png` | Cosine improves to about 0.50, showing partial alignment. |
| `semantic_loss.png` | Semantic loss decreases, so corrected loss is working. |
| `decoder_ce.png` | Validation CE increases while train CE decreases, showing decoder overfitting. |
| `ae_ce.png` | Decoder anchor helps train loss but does not prevent validation overfitting. |
| `cosine_loss.png` | `1 - cosine(z,s)` decreases, so embedding direction improves. |

### Conclusion

The corrected loss improves embedding alignment, but translation is still weak because the decoder overfits and collapses to common weather sentences.
